### ISSUE TITLE
How to make ocamlformat use `.ocamlformat` file from a specific directory?

### ISSUE THREAD

**Issue by utkarshkukreti** [ocaml-ppx/ocamlformat#573]
## Context

I wrote the initial ocamlformat <-> atom-beautify integration [a while back](https://github.com/Glavin001/atom-beautify/pull/2207). I'm trying to make it support reading `.ocamlformat` config from the project folder.

## Problem

atom-beautify (with beautify on save mode) formats the file _before_ it's saved to disk. The implementation above saves the contents of the file to a temporary file, calls `ocamlformat` binary on it, and then sends the output back to the editor before the file is saved. This method does not read the correct `.ocamlformat` file because the temporary file is not stored in the same directory as the original file.

## Question

Is it possible to specify which directory to scan for the `.ocamlformat` file while specifying a different file as input?

I tried passing `--root` but it doesn't work because passing a root argument presumably enables `--disable-outside-detected-project` automatically:

```
$ ocamlformat --root=/path/to/original/file's/directory /tmp/a.ml
File "/tmp/a.ml":
Warning: Ocamlformat disabled because [--disable-outside-detected-project] was given and no project root was found
...
```

**Comment by jberdine** [comment]
Yes, using `--name` should work, see https://github.com/ocaml-ppx/ocamlformat/blob/master/ocamlformat-help.txt#L250 .

And thanks for the atom-beautify integration!

**Comment by jberdine** [comment]
@utkarshkukreti I failed to mention this before, but please feel free to send a PR with an addition to the README.md file that points to your atom-beautify integration.

**Comment by Wilfred** [comment]
I hit this issue today, and ended up using `--name` rather than `--root`. I'm struggling to find a use case for `--root`.
